### PR TITLE
feat: gallery ui revamp

### PR DESCRIPTION
**Pull request by DarkPhoenix2704**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)
With Image
image

Without Image
image

Cards If no Image
image

### CODE REVIEW

**Comment by dstala** [comment]
- [ ] Left padding for content text
- [ ] Indicator dots below image
- [ ] Image line
- [ ] Empty cells- column header
- [ ] Every field type is left aligned

Screenshot 2023-09-20 at 9 59 26 AM

File: packages/nc-gui/components/smartsheet/Gallery.vue [hunk 1]
```diff
@@ -263,18 +269,26 @@ watch(
                   </template>
 
                   <template #prevArrow>
-                    <div style="z-index: 1"></div>
+                    <div style="z-index: 1">
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/Gallery.vue:272]: don't use `style` and use class util instead

File: packages/nc-gui/components/smartsheet/Gallery.vue [hunk 2]
```diff
@@ -263,18 +269,26 @@ watch(
...
+                      <MdiChevronLeft
+                        class="text-gray-700 w-6 h-6 absolute left-1.5 bottom-[-90px] !opacity-0 !group-hover:opacity-100 !bg-white border-1 border-gray-200 rounded-md transition"
+                      />
+                    </div>
                   </template>
 
                   <template #nextArrow>
-                    <div style="z-index: 1"></div>
+                    <div style="z-index: 1">
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/Gallery.vue:280]: same as above

File: packages/nc-gui/components/smartsheet/Gallery.vue [hunk 3]
```diff
@@ -353,50 +390,53 @@ watch(
...
+  @apply !p-0;
 }
 
 :deep(.slick-dots li button) {
   background-color: black;
 }
 
 .ant-carousel.gallery-carousel :deep(.slick-dots) {
-  position: relative;
+  @apply !w-auto;
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/Gallery.vue:405]: All css should be written in class util format. e.g. `position: absolute` -> `absolute`, `height: auto` -> `h-auto` ... 

In this way, something like below is expected:

```
@apply !w-auto absolute h-auto bottom-[3px];
```

File: packages/nc-gui/components/smartsheet/Gallery.vue [hunk 4]
```diff
@@ -283,21 +297,44 @@ watch(
...
+                  :edit-enabled="false"
+                  :read-only="true"
+                />
+              </h2>
 
               <div v-for="col in fieldsWithoutCover" :key="`record-${record.row.id}-${col.id}`">
                 <div
                   v-if="!isRowEmpty(record, col) || isLTAR(col.uidt, col.colOptions)"
-                  class="flex flex-col space-y-1 px-4 mb-6 bg-gray-50 rounded-lg w-full"
+                  class="flex flex-col ml-[-4px] rounded-lg w-full"
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/Gallery.vue:322]: may double think about if negative margin is really necessary. most of the time people just misuse it to fix the incorrect layout. it's a bad practice.

File: packages/nc-gui/components/smartsheet/Gallery.vue [hunk 5]
```diff
@@ -59,6 +59,8 @@ provide(IsFormInj, ref(false))
 provide(IsGalleryInj, ref(true))
 provide(IsGridInj, ref(false))
 
+provide(RowHeightInj, ref(1 as const))
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/Gallery.vue:62]: is `as const` required` here?

File: packages/nc-gui/components/smartsheet/Gallery.vue [hunk 6]
```diff
@@ -353,50 +390,53 @@ watch(
 <style scoped>
 .nc-gallery-container {
   grid-auto-rows: 1fr;
-  grid-template-columns: repeat(auto-fit, minmax(230px, 1fr));
+  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
+}
+
+.ant-card-body {
+  @apply !p-0;
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nc-gui/components/smartsheet/Gallery.vue:397]: removed✅

File: packages/nc-gui/components/smartsheet/Gallery.vue (hunk 5 above)
Comment (DarkPhoenix2704) [reply packages/nc-gui/components/smartsheet/Gallery.vue:62]: image
Yes. The possible values are 1, 2, 4 and 6

File: packages/nc-gui/components/smartsheet/Gallery.vue (hunk 4 above)
Comment (DarkPhoenix2704) [reply packages/nc-gui/components/smartsheet/Gallery.vue:322]: updated

**Review by dstala — APPROVED** [review]

**Pull request merged**